File: Projects/rosalind/SSEQ.java
import java.io.*;
import java.util.*;

public class SSEQ {
	public static void main(String[] args) throws IOException {
		Scanner in = new Scanner(System.in);
		List<String> reads = new ArrayList<>();
		String read = null;
		while (true) {
			if (!in.hasNext()) {
				reads.add(read);
				break;
			}
			String line = in.next();
			if (line.charAt(0) == '>') {
				if (read != null)
					reads.add(read);
				read = "";
			} else
				read += line;
		}
		String seq = reads.get(0), pat = reads.get(1);
		int i = 0, j = 0;
		for (; j < pat.length(); j++)
			for (; i < seq.length(); i++)
				if (seq.charAt(i) == pat.charAt(j)) {
					System.out.println(++i);
					break;
				}
		in.close();
	}
}

File: Projects/rosalind/SSEQ.kt
package rosalind

import java.io.IOException
import java.util.*

object SSEQ {
    @Throws(IOException::class)
    @JvmStatic
    fun main(args: Array<String>) {
        val stdin = Scanner(System.`in`)
        val reads: MutableList<String?> = ArrayList()
        var read: String? = null
        while (true) {
            if (!stdin.hasNext()) {
                reads.add(read)
                break
            }
            val line = stdin.next()
            if (line[0] == '>') {
                if (read != null) reads.add(read)
                read = ""
            } else read += line
        }
        val seq = reads[0]
        val pat = reads[1]
        var i = 0
        var j = 0
        while (j < pat!!.length) {
            while (i < seq!!.length) {
                if (seq[i] == pat[j]) {
                    println(++i)
                    break
                }
                i++
            }
            j++
        }
        stdin.close()
    }
}